packages/node/src/node.ts

```diff
@@ -55,6 +56,21 @@ export class Node {
       `${nodeConfig.STORE_KEY_PREFIX}/${this.signer.address}`
     );
     this.instructionExecutor = new InstructionExecutor(networkContext);
+    this.instructionExecutor.middlewares.add(
+      Opcode.STATE_TRANSITION_COMMIT,
+      async (message, next, context) => {
+        const { appIdentityHash } = message.params as any; // TODO: not every params has an appIdentityHash field
+        const transaction = context.commitment!.transaction([
```

Comment:
If you do type checking you won't need to add `!`

Reply:
Actually you're right; but it means we should be checking if the value is undefined instead of muting the type checker probably